include/irr/builtin/material_compiler/glsl/common.glsl

```diff
@@ -330,14 +340,17 @@ void setCurrInteraction(in vec3 N)
 	irr_glsl_IsotropicViewSurfaceInteraction interaction = irr_glsl_calcFragmentShaderSurfaceInteraction(campos, WorldPos, N);
 	currInteraction = irr_glsl_calcAnisotropicInteraction(interaction);
 }
-void setCurrInteraction(in vec3 N, in bool ts)
+void setCurrInteraction(in MC_precomputed_t precomp)
+{
+	setCurrInteraction(precomp.N);
+}
+void updateCurrInteraction(in MC_precomputed_t precomp, in vec3 N, in bool ts)
 {
 #ifdef NO_TWOSIDED
-	setCurrInteraction(N);
+	setCurrInteraction(precomp);
 #else
 	vec3 campos = irr_glsl_MC_getCamPos();
-	vec3 view = campos - WorldPos;
-	setCurrInteraction((ts && dot(N,view)<0.0) ? -N : N);
+	setCurrInteraction((ts && precomp.NdotV<0.0) ? -N : N);
```

Comment:
this is a mess it still doesnt do what its supposed to
https://github.com/buildaworldnet/IrrlichtBAW/pull/424#discussion_r509456818

you shouldn't even have to pass the Normal to this function.

Reply:
?? well, obviously i need to normal to set surface interaction

Reply:
you no longer need to flip, if you're doing twosided as separate instruction streams, so this function needs not to exist